extensions/wikia/AbPerformanceTesting/js/FrontendDelay.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * Introduce a delayed onDOMReady event to measure the impact of performance regressions
+ *
+  * @see FrontendDelay.class.php
+ */
+(function(w, log, weppy, $) {
+
+	/**
+	 * @see https://github.com/jfriend00/docReady
+	 */
+	var docReady = (function() {
+		"use strict";
+		var readyList = [];
+		var readyFired = false;
+		var readyEventHandlersInstalled = false;
+
+		// call this when the document is ready
+		// this function protects itself against being called more than once
+		function ready() {
+			if (!readyFired) {
+				// this must be set to true before we start calling callbacks
+				readyFired = true;
+				for (var i = 0; i < readyList.length; i++) {
+					// if a callback here happens to add new ready handlers,
+					// the docReady() function will see that it already fired
+					// and will schedule the callback to run right after
+					// this event loop finishes so all handlers will still execute
+					// in order and no new ones will be added to the readyList
+					// while we are processing the list
+					readyList[i].fn.call(window, readyList[i].ctx);
+				}
+				// allow any closures held by these functions to free
+				readyList = [];
+			}
+		}
+
+		function readyStateChange() {
+			if ( document.readyState === "complete" ) {
+				ready();
+			}
+		}
+
+		// This is the one public interface
+		// docReady(fn, context);
+		// the context argument is optional - if present, it will be passed
+		// as an argument to the callback
+		return function(callback, context) {
+			// if ready has already fired, then just schedule the callback
+			// to fire asynchronously, but right away
+			if (readyFired) {
+				setTimeout(function() {callback(context);}, 1);
```

Comment:
No idea about the reason. It's a 3rd-party code.

`setTimeout` with 0 is usually delayed by ~20ms. 0 or 1 - should not really make any difference